Update the versalnet apu to load TF-A as APU runs at EL1 NS

kedareswararao opened the pull request:
This pull request updates the versalnet_apu board support to build TF-A alongside the Zephyr application.
By default, the APU runs at EL1-Non Secure, and QEMU requires a secure OS to configure GIC-V3 and other secure settings. Integrating TF-A with the Zephyr build ensures the necessary secure initialization is performed before the Zephyr application runs.

Commit:
boards: amd: versalnet_apu: Add comprehensive TF-A build integration

Enable ARM Trusted Firmware support for Versal NET APU board with
platform-specific optimizations.

Changes:
- Add TF-A platform configuration (PLAT=versal_net)
- Configure board-specific TF-A parameters (TFA_NO_PM, PRELOADED_BL33_BASE)
- Add automatic TF-A module selection in Kconfig
- Update QEMU kernel loading to use TF-A bl31.elf
- Modified data address configuration to match QEMU implementation

The board now automatically builds ARM Trusted Firmware during Zephyr
application compilation with Versal NET specific optimizations.

Boot flow: Boot PDI -> TF-A -> Zephyr

Signed-off-by: Appana Durga Kedareswara rao <appana.durga.kedareswara.rao@amd.com>

michalsimek reviewed (commented):
I am expecting that commit message say some information about s/3DD/3EE/ qemu command line change.

kedareswararao commented on boards/amd/versalnet_apu/board.cmake:
```diff
@@ -12,11 +12,24 @@ set(QEMU_CPU_TYPE_${ARCH} cortexa78)
+if(CONFIG_BUILD_WITH_TFA)
+  set(TFA_PLAT "versal_net")
+  # Add Versal NET specific TF-A build parameters
+  set(TFA_EXTRA_ARGS "TFA_NO_PM=1;PRELOADED_BL33_BASE=0x0")
```
Thanks for the review fixed in the updated pull request

kedareswararao commented:
Thanks @JarmouniA  for assigning it to Michal.
This pull request was created a couple of months ago, and since there were merge conflicts, I’ve rebased it on the latest master.
@JarmouniA @henrikbrixandersen @ibirnbaum @michalsimek @wearyzen 
Could you please review it, and if everything looks good, please approve it so that Michal can proceed with the merge?”

Commit:
modules: tf-a: Add extensible board-specific build parameter support

Enhance the TF-A build system to support board-specific build parameters
through TFA_EXTRA_ARGS mechanism.

Changes:
- Add TFA_EXTRA_ARGS variable support for board-specific parameters
- Implement proper string-to-list conversion for make argument expansion

This allows individual board configurations to specify platform-specific
TF-A build parameters without modifying the core TF-A build logic.

Signed-off-by: Appana Durga Kedareswara rao <appana.durga.kedareswara.rao@amd.com>

michalsimek reviewed (approved):
I have retest it and works for me.

NOTICE:  TF-A running on QEMU 1.4
NOTICE:  BL31: Secure code at 0x60000000
NOTICE:  BL31: Non secure code at 0x0
NOTICE:  BL31: v2.13.0(release):v2.13.0-2-g0a29cac8fe0f
NOTICE:  BL31: Built : 11:52:07, Nov 28 2025
*** Booting Zephyr OS build v4.3.0-1174-g92210c1e9cdc ***
Hello World! versalnet_apu/amd_versalnet_apu
